packages/design-system/src/components/date-field/date-field.tsx

```diff
@@ -0,0 +1,171 @@
+import { bodies } from '../../styles';
+import { callRenderProps, inlineVars, mergeClassNames } from '../../utils';
+import {
+  AriaFieldErrorContext,
+  type AriaLabelContext,
+  AriaTextContext,
+} from '../aria';
+import { DateInputContext, type DateInputProps } from '../date-input';
+import { IconContext, type IconProps } from '../icon';
+import { dateFieldClassNames, dateFieldStateVars } from './date-field.css';
+import type {
+  DateFieldMapping,
+  DateFieldProps,
+  DateFieldRenderProps,
+} from './types';
+
+const defaultMapping: DateFieldMapping = {
+  description: {
+    sm: bodies.xs,
+    lg: bodies.xs,
+  },
+  error: {
+    sm: bodies.xs,
+    lg: bodies.xs,
+  },
+  icon: {
+    sm: { size: 'xs' },
+    lg: { size: 'md' },
+  },
+};
+
+const defaultSize = 'lg';
+
+export const DateFieldContext =
+  createContext<ContextValue<DateFieldProps<DateValue>, HTMLDivElement>>(null);
+
+export const DateField = forwardRef(function DateField<T extends DateValue>(
+  props: DateFieldProps<T>,
+  ref: ForwardedRef<HTMLDivElement>,
+) {
+  [props, ref] = useContextProps(props, ref, DateFieldContext);
+
+  props = useDefaultProps(
+    props as DateFieldProps<DateValue>,
+    'DateField',
+  ) as DateFieldProps<T>;
+
+  const {
+    children: childrenProp,
+    classNames: classNamesProp,
+    mapping: mappingProp,
+    size = defaultSize,
+    value,
+    ...rest
+  } = props;
+
+  const theme = useTheme();
+
+  const mapping = useMemo(
+    () => ({
+      ...defaultMapping,
+      ...mappingProp,
+    }),
+    [mappingProp],
+  );
+
+  const classNames = useMemo(
+    () =>
+      mergeClassNames(dateFieldClassNames, theme.DateField, classNamesProp, {
+        description: mapping.description[size],
+        error: mapping.error[size],
+      }),
+    [theme.DateField, classNamesProp, mapping, size],
+  );
+
+  const style = useCallback(
+    (renderProps: DateFieldRenderProps) =>
+      inlineVars(dateFieldStateVars, {
+        ...renderProps,
+        size,
+      }),
+    [size],
+  );
+
+  const values = useMemo<
+    [
+      [typeof DateInputContext, ContextValue<DateInputProps, HTMLDivElement>],
+      [typeof AriaLabelContext, ContextValue<LabelProps, HTMLLabelElement>],
+      [typeof AriaTextContext, ContextValue<TextProps, HTMLElement>],
+      [
+        typeof AriaFieldErrorContext,
+        ContextValue<FieldErrorProps, HTMLElement>,
+      ],
+      [typeof IconContext, ContextValue<IconProps, HTMLDivElement>],
+    ]
+  >(
+    () => [
+      [DateInputContext, { classNames: classNames?.input, size }],
+      [LabelContext, { className: classNames?.label }],
+      [
+        AriaTextContext,
+        {
+          slots: {
+            description: { className: classNames?.description },
+          },
+        },
+      ],
+      [AriaFieldErrorContext, { className: classNames?.error }],
+      [IconContext, { ...mapping.icon[size], classNames: classNames?.icon }],
+    ],
+    [classNames, size],
```

Comment:
missing useMemo dependency
```suggestion
    [classNames, size, mapping.icon[size]],
```